File: objc-1/f10.super/Student.m
#import "Student.h"

@implementation Student

// ###### Constructors ######
- (Student *) init: (const char *) n gpa:(double) g {
    self = [super init: n];
    if (self) { [self gpa: g]; }
    return self;
}

// ###### Mutators ######
- (void) gpa: (double) g { gpa = g; }

// ###### Accessors ######
- (double) gpa { return gpa; }

@end

File: java/g10.super/Student.java
public class Student extends Person
{
    private Double gpa;

    // Constructors
    public Student(String name)              { super(name); }
    public Student(String name, Double gpa)  { super(name); this.setGPA(gpa); }

    // Mutators - functions that set stuff
    public void   setGPA(Double gpa) { this.gpa = gpa; }

    // Accessors - functions that get stuff
    public Double getGPA()           { return gpa; }
}
